cmd/promtool/unittest.go

```diff
@@ -311,12 +311,8 @@ func (tg *testGroup) test(testname string, evalInterval time.Duration, groupOrde
 			return errs
 		}
 
-		for {
-			if !(curr < len(alertEvalTimes) && ts.Sub(mint) <= time.Duration(alertEvalTimes[curr]) &&
-				time.Duration(alertEvalTimes[curr]) < ts.Add(evalInterval).Sub(mint)) {
-				break
-			}
-
+		for curr < len(alertEvalTimes) && ts.Sub(mint) <= time.Duration(alertEvalTimes[curr]) &&
+			time.Duration(alertEvalTimes[curr]) < ts.Add(evalInterval).Sub(mint) {
```

Comment:
This feels wrong, shouldn't we negate the latter part too (and replace && with ||)? 

> time.Duration(alertEvalTimes[curr]) < ts.Add(evalInterval).Sub(mint)

Also do you mind removing `<` when we compare times and use time `Before` and `After` methods?

Reply:
Ah lol, ok I see my mistake I somehow read `!(curr < len(alertEvalTimes) && ts.Sub(mint) <= time.Duration(alertEvalTimes[curr]) && ..... )` as `!(curr < len(alertEvalTimes) && ts.Sub(mint) <= time.Duration(alertEvalTimes[curr])) && ..... ` (parentesis of `len` and `time.Duration(..)` misled me), sorry for the fake alarm!

For After/Before, you can still represent this with After/Before as we DO use `ts` initially, so we could operate on time.Time e.g. 

```
ts.Sub(mint) <= time.Duration(alertEvalTimes[curr])
```

is the same to sth like:

```
ts.Before(mint.Add(alertEvalTimes[curr])).
```

Not a blocker, though, not sure what's more readable 🤷🏽